### PR TITLE
API-369-370-371: add tests for variant product API

### PR DESCRIPTION
**Pull request by solivier**
Add a test for variant product (on the API) it checks the parent field is well set.

### CODE REVIEW

File: src/Pim/Bundle/ApiBundle/tests/integration/Controller/Product/DeleteVariantProductIntegration.php [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+class DeleteVariantProductIntegration extends AbstractProductTestCase
+{
+    public function testDeleteAVariantProduct()
+    {
+        $client = $this->createAuthenticatedClient();
+
+        $this->assertCount(247, $this->get('pim_catalog.repository.product')->findAll());
+
+        $fooProduct = $this->get('pim_catalog.repository.product')->findOneByIdentifier('apollon_blue_xl');
+        $this->get('pim_catalog.elasticsearch.indexer.product')->index($fooProduct);
```
Comment (skeleton) [inline src/Pim/Bundle/ApiBundle/tests/integration/Controller/Product/DeleteVariantProductIntegration.php]: why do you need it?

File: src/Pim/Bundle/ApiBundle/tests/integration/Controller/Product/DeleteVariantProductIntegration.php [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+class DeleteVariantProductIntegration extends AbstractProductTestCase
+{
+    public function testDeleteAVariantProduct()
+    {
+        $client = $this->createAuthenticatedClient();
+
+        $this->assertCount(247, $this->get('pim_catalog.repository.product')->findAll());
+
+        $fooProduct = $this->get('pim_catalog.repository.product')->findOneByIdentifier('apollon_blue_xl');
```
Comment (skeleton) [inline src/Pim/Bundle/ApiBundle/tests/integration/Controller/Product/DeleteVariantProductIntegration.php]: not foo ;)

File: src/Pim/Bundle/ApiBundle/tests/integration/Controller/Product/DeleteVariantProductIntegration.php [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $product = $this->get('pim_catalog.builder.variant_product')->createProduct('apollon_blue_xl', 'clothing');
```
Comment (skeleton) [inline src/Pim/Bundle/ApiBundle/tests/integration/Controller/Product/DeleteVariantProductIntegration.php]: why do you need to create an other product as you already have 247 products in your catalog?

File: src/Pim/Bundle/ApiBundle/tests/integration/Controller/Product/GetVariantProductIntegration.php [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $product = $this->get('pim_catalog.builder.variant_product')->createProduct('apollon_blue_xl', 'clothing');
```
Comment (skeleton) [inline src/Pim/Bundle/ApiBundle/tests/integration/Controller/Product/GetVariantProductIntegration.php]: same remark

File: tests/catalog/technical/attributes.csv [hunk 5]
```diff
@@ -26,3 +26,5 @@ a_scopable_image;;;;;;;;attributeGroupB;0;;;;;;;;;1;0;pim_catalog_image;0;0;;;
 a_localizable_scopable_image;;;;;;;;attributeGroupB;1;;;;;;;;;1;0;pim_catalog_image;0;0;;;
 a_scopable_price;;;;;;1;;attributeGroupA;0;;;;;;;;;1;9;pim_catalog_price_collection;0;0;;;
 a_localized_and_scopable_text_area;;;;;;;;attributeGroupA;1;;;;;;;;;1;10;pim_catalog_textarea;0;0;;;0
+color;;1;;;;;;attributeGroupB;0;;;;;;;;;0;0;pim_catalog_simpleselect;0;0;;;
```
Comment (skeleton) [inline tests/catalog/technical/attributes.csv]: a_simple_select_color & a_simple_select_size ?

File: src/Pim/Bundle/ApiBundle/tests/integration/Controller/Product/DeleteVariantProductIntegration.php (hunk 2 above)
Comment (solivier) [reply src/Pim/Bundle/ApiBundle/tests/integration/Controller/Product/DeleteVariantProductIntegration.php]: just copied and paste the DeleteProductIntegration :p but ill change :)

File: src/Pim/Bundle/ApiBundle/tests/integration/Controller/Product/DeleteVariantProductIntegration.php (hunk 1 above)
Comment (solivier) [reply src/Pim/Bundle/ApiBundle/tests/integration/Controller/Product/DeleteVariantProductIntegration.php]: it has been added in the DeleteProductIntegration with the commit ("fix integration tests") I can try to remove it.

**Pull request merged**